statsmodels/stats/multitest.py

```diff
@@ -444,3 +444,237 @@ def fdrcorrection_twostage(pvals, alpha=0.05, method='bky', iter=False,
+
+    >>> fdr = local_fdr(zscores)
+
+    Use an empirical null distribution estimated from the data:
+
+    >>> null = EmpiricalNull(zscores)
+    >>> fdr = local_fdr(zscores, null_density=null.pdf)
+    """
+
+    from statsmodels.genmod.generalized_linear_model import GLM
+    from statsmodels.genmod.generalized_linear_model import families
+    from statsmodels.regression.linear_model import OLS
+
+    # Bins for Poisson modeling of the marginal Z-score density
+    minz = min(zscores)
+    maxz = max(zscores)
+    bins = np.linspace(minz, maxz, nbins)
+
+    # Bin counts
+    zhist = np.histogram(zscores, bins)[0]
+
+    # Bin centers
+    zbins = (bins[:-1] + bins[1:]) / 2
+
+    # The design matrix at bin centers
+    dmat = np.vander(zbins, deg + 1)
+
+    # Use this to get starting values for Poisson regression
+    md = OLS(np.log(1 + zhist), dmat).fit()
+
+    # Poisson regression
+    md = GLM(zhist, dmat, family=families.Poisson()).fit(start_params=md.params)
+
+    # The design matrix for all Z-scores
+    dmat_full = np.vander(zscores, deg + 1)
+
+    # The height of the estimated marginal density of Z-scores,
+    # evaluated at every observed Z-score.
+    fz = md.predict(dmat_full) / (len(zscores) * (bins[1] - bins[0]))
+
+    # The null density.
+    if null_density is None:
+        f0 = np.exp(-0.5 * zscores**2) / np.sqrt(2 * np.pi)
+    else:
+        f0 = null_density(zscores)
+
+    # The local FDR values
+    fdr = p0 * f0 / fz
+
+    fdr = np.clip(fdr, 0, 1)
+
+    return fdr
+
+
+class EmpiricalNull(object):
+    """
+    Estimate empirical distribution of null Z-scores.
+
+    The observed Z-scores consist of both null and non-null values.
+    The fitted distribution is Gaussian, but may have non-zero mean
+    and/or non-unit scale.
+
+    Parameters
+    ----------
+    zscores : array-like
+        The observed Z-scores.
+    null_lb : float
+        Z-scores between `null_lb` and `null_lb` are all considered to be
+        true null hypotheses.
+    null_ub : float
+        See `null_lb`.
+    estimate_mean : bool
+        If True, estimate the mean of the distribution.  If False, the
+        mean is fixed at zero.
+    estimate_scale : bool
+        If True, estimate the scale of the distribution.  If False, the
+        scale parameter is fixed at 1.
+    estimate_prob : bool
+        If True, estimate the proportion of true null hypotheses (i.e.
+        the proportion of z-scores with expected value zero).  If False,
+        the probability parameter is fixed at 1.
+
+    Returns
+    -------
+    mean0 : float
+        The estimated mean of the empirical null distribution
+    scale0 : float
+        The estimated standard deviation of the empirical null distribution
+    prob0 : float
+        The estimated proportion of true null hypotheses
+    null_density : function (from reals to positive reals)
+        The fitted empirical null density function
```

Comment:
AFAICS this is not an attribute but the `pdf` method which will be automatically documented (methods for classes)